x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/rules_table_utility_bar.test.tsx

```diff
@@ -9,52 +9,72 @@ import React from 'react';
 import { mount } from 'enzyme';
 import { waitFor } from '@testing-library/react';
 
-import { AllRulesUtilityBar } from './utility_bar';
+import { AllRulesUtilityBar } from './rules_table_utility_bar';
 import { TestProviders } from '../../../../../common/mock';
 
+jest.mock('./rules_table/rules_table_context');
+
 describe('AllRules', () => {
   it('renders AllRulesUtilityBar total rules and selected rules', () => {
     const wrapper = mount(
       <TestProviders>
         <AllRulesUtilityBar
           canBulkEdit
           onRefresh={jest.fn()}
-          paginationTotal={4}
+          pagination={{
+            page: 1,
+            perPage: 10,
+            total: 21,
+          }}
           numberSelectedItems={1}
           onGetBulkItemsPopoverContent={jest.fn()}
           isAutoRefreshOn={true}
           onRefreshSwitch={jest.fn()}
-          hasBulkActions
         />
       </TestProviders>
     );
 
-    expect(wrapper.find('[data-test-subj="showingRules"]').at(0).text()).toEqual('Showing 4 rules');
+    expect(wrapper.find('[data-test-subj="showingRules"]').at(0).text()).toEqual(
+      'Showing 1-10 of 21 rules'
+    );
     expect(wrapper.find('[data-test-subj="selectedRules"]').at(0).text()).toEqual(
       'Selected 1 rule'
     );
   });
 
-  it('does not render total selected and bulk actions when "hasBulkActions" is false', () => {
-    const wrapper = mount(
-      <TestProviders>
-        <AllRulesUtilityBar
-          canBulkEdit
-          onRefresh={jest.fn()}
-          paginationTotal={4}
-          numberSelectedItems={1}
-          onGetBulkItemsPopoverContent={jest.fn()}
-          isAutoRefreshOn={true}
-          onRefreshSwitch={jest.fn()}
-          hasBulkActions={false}
-        />
-      </TestProviders>
-    );
-
-    expect(wrapper.find('[data-test-subj="showingRules"]').exists()).toBeFalsy();
-    expect(wrapper.find('[data-test-subj="tableBulkActions"]').exists()).toBeFalsy();
-    expect(wrapper.find('[data-test-subj="showingExceptionLists"]').at(0).text()).toEqual(
-      'Showing 4 lists'
+  describe('renders correct pagination label when', () => {
+    it.each([
+      [0, 0, 0, 'Showing 0-0 of 0 rules'],
+      [1, 1, 1, 'Showing 1-1 of 1 rule'],
+      [1, 10, 21, 'Showing 1-10 of 21 rules'],
+      [1, 10, 8, 'Showing 1-8 of 8 rules'],
+      [2, 10, 31, 'Showing 11-20 of 31 rules'],
+      [4, 10, 31, 'Showing 31-31 of 31 rules'],
+      [1, 5, 4, 'Showing 1-4 of 4 rules'],
+      [1, 100, 100, 'Showing 1-100 of 100 rules'],
+      [2, 100, 101, 'Showing 101-101 of 101 rules'],
+    ])(
```

Comment:
I wonder if we need all of these test cases, do they all contribute to the coverage? Maybe it would be possible to ensure coverage with fewer cases.

Reply:
Okay, I think the fewer tests we have, the better (without harming coverage). So if we could get rid of three extra test cases, that's already something.

Also, if you decide to stick with `test.each`, it would be great to add some comments to the cases (as you did in the comment above) to clarify what they are supposed to test. Without comments, it's a bit hard to figure out the meaning of tests by looking at the input args only.